From ceb69fe6f7ce6ee14fae33a060891b125223c125 Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Sat, 20 May 2017 13:40:41 -0700
Subject: [PATCH] Add a target that just build packages CLI needs

This cuts down on build time and also reduces the set of dependencies
we need to pull in (since we aren't building code we arne't going to
use or running tests).
---
 build.proj | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/build.proj b/build.proj
index 3b09fa1..a8ef5c0 100644
--- a/build.proj
+++ b/build.proj
@@ -67,6 +67,42 @@
     </TestProjects>
   </PropertyGroup>
 
+  <Target Name="BuildForCli">
+    <MakeDir Directories="$(ArtifactsDir);$(PackageOutputPath);$(TemplatesOutputPath);$(DevPath)" />
+    <CallTarget Targets="CollectGitInfo" />
+
+    <!-- Build the inner ring (abstractions, core, core contracts, etc.) -->
+    <MSBuild Projects="$(InnerRingProjects)" Targets="Restore;Build"
+             Properties="TargetFramework=$(InnerRingTargetFramework);Configuration=$(Configuration)"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <MSBuild Projects="$(InnerRingProjects)" Targets="Pack"
+             Properties="TargetFrameworks=$(InnerRingPackTargetFrameworks);Configuration=$(Configuration);PackageOutputPath=$(PackageOutputPath);NoBuild=true;IncludeSymbols=true"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <!-- Build Edge (the only outerring component that CLI depends on) -->
+    <MSBuild Projects="src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj" Targets="Restore;Build"
+             Properties="TargetFramework=$(OuterRingTargetFramework);Configuration=$(Configuration)"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <MSBuild Projects="src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj" Targets="Pack"
+             Properties="TargetFrameworks=$(OuterRingPackTargetFrameworks);Configuration=$(Configuration);PackageOutputPath=$(PackageOutputPath);NoBuild=true;IncludeSymbols=true"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <!-- Build projects that only build for the dotnet CLI -->
+    <MSBuild Projects="$(CliProjects)" Targets="Restore;Build"
+             Properties="TargetFramework=$(OuterRingTargetFramework);Configuration=$(Configuration);PackageOutputPath=$(PackageOutputPath);NoBuild=true"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <MSBuild Projects="$(CliProjects)" Targets="Pack"
+             Properties="TargetFramework=$(OuterRingTargetFramework);Configuration=$(Configuration);PackageOutputPath=$(PackageOutputPath);NoBuild=true;IncludeSymbols=true"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+
+    <!-- Build templates -->
+    <MSBuild Projects="template_feed/Template.proj" Targets="Build"
+             RunEachTargetSeparately="true" StopOnFirstFailure="true" />
+  </Target>
+
   <Target Name="Build">
 
     <MakeDir Directories="$(ArtifactsDir);$(PackageOutputPath);$(TemplatesOutputPath);$(DevPath)" />
-- 
1.8.3.1

